src/main/java/core/basesyntax/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.Storage;
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void handle(Storage storage, FruitTransaction transaction) {
+        storage.reduceFruitQuantity(transaction.getFruit(), transaction.getQuantity());
```

Comment:
You need to check if the result balance is positive before saving it in the Storage. If the balance is negative, you should throw a RuntimeException. This is a requirement from the checklist: 'Handling Purchase operation. Check result balance before saving it in the Storage - it should be positive. Throw RuntimeException in case the balance is negative.'